sphinx.ext.pngmath has been deprecated since 1.4 and removed in 1.8. It has been replaced by imgmath

--- doc/sphinx/conf.py.in
+++ doc/sphinx/conf.py.in
@@ -12,6 +12,7 @@
 # serve to show the default.

 import sys, os
+from sphinx import version_info as sphinx_version

 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
@@ -23,7 +24,8 @@ sys.path = [os.path.abspath('@CMAKE_SOURCE_DIR@/src')]+sys.path

 # Add any Sphinx extension module names here, as strings. They can be extensions
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
-extensions = ['sphinx.ext.autodoc', 'sphinx.ext.doctest', 'sphinx.ext.intersphinx', 'sphinx.ext.todo', 'sphinx.ext.pngmath']
+extensions = ['sphinx.ext.autodoc', 'sphinx.ext.doctest', 'sphinx.ext.intersphinx', 'sphinx.ext.todo']
+extensions.append('sphinx.ext.pngmath' if sphinx_version < (1, 4) else 'sphinx.ext.imgmath')

 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
